src/lib/util/statistics.ts

```diff
@@ -0,0 +1,412 @@
+import {
+  QuestionnaireByID,
+  ResponseResult,
+  QuestionDetails,
+  ResponseBody,
+  QuestionType
+} from '/@/lib/apis'
+
+export const adjustQuestions = (
+  questionnaire: QuestionnaireByID,
+  questions: QuestionDetails[],
+  results: ResponseResult[]
+): ResultsPerQuestion => {
+  const resQuestions: BaseTypeQuestion[] = questions.map(q => {
+    const resultsPerQuestionWithUser = results
+      .map(r =>
+        r.body.map(body => ({
+          ...body,
+          modified_at: r.modified_at,
+          submitted_at: r.submitted_at,
+          traqID: r.traqID
+        }))
+      )
+      .map(bodies => bodies.find(v => v.questionID === q.questionID))
+      .filter((v): v is Exclude<typeof v, undefined> => !!v)
+    return {
+      type: q.question_type,
+      question: q,
+      results: resultsPerQuestionWithUser
+    }
+  })
+  return {
+    questionnaire: questionnaire,
+    questions: resQuestions
+  }
+}
+
+//テキスト
+export interface TextTypeQuestion {
+  type: 'Text'
+  question: Omit<
+    QuestionDetails,
+    | 'questionnaireID'
+    | 'max_bound'
+    | 'min_bound'
+    | 'options'
+    | 'regex_pattern'
+    | 'scale_label_left'
+    | 'scale_label_right'
+    | 'scale_max'
+    | 'scale_min'
+  >
```

Comment:
```suggestion
  question: Omit<
    QuestionDetails,
    | 'questionnaireID'
    | 'max_bound'
    | 'min_bound'
    | 'options'
    | 'regex_pattern'
    | 'scale_label_left'
    | 'scale_label_right'
    | 'scale_max'
    | 'scale_min'
  > &
      QuestionDetails
```
ってするとうまくいきます

他も同様に

Reply:
これ完全に嘘です